controllers/record-utils.js

```diff
@@ -20,17 +22,19 @@ var request = function(opts, callback) {
 };
 
 var createByForm = function(data, callback) {
-  var required = ['message', 'from'],
-      optional = ['reported_date', 'locale', 'gateway_ref'];
-  for (var k in required) {
-    if (required.hasOwnProperty(k)) {
-      if (!exists(data[required[k]])) {
-        return callback(new Error('Missing required field: ' + required[k]));
-      }
-    }
+  if (empty(data.from)) {
+    return callback(new Error('Missing required field: from'));
```

Comment:
I was wondering if there was a way to communicate that we needed a value for that field, not just the presence of the field.  Possibly worthwhile as this is the cause of the bug...